src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -835,6 +855,43 @@ uct_ib_mlx5_devx_mem_reg_gva(uct_md_h uct_md, unsigned flags, uct_mem_h *memh_p)
     return status;
 }
 
+static ucs_status_t
+uct_ib_mlx5_devx_mem_window_reg(uct_md_h uct_md,
+                                const uct_md_mem_reg_params_t *params,
+                                uct_mem_h *memh_p)
+{
+    uct_ib_mlx5_md_t *md         = ucs_derived_of(uct_md, uct_ib_mlx5_md_t);
+    uct_ib_mlx5_devx_mem_t *base =
+                    UCT_MD_MEM_REG_FIELD_VALUE(params, memh, FIELD_MEMH, NULL);
+    uct_ib_mlx5_devx_mem_t *memh;
+    ucs_status_t status;
+
+    if (ENABLE_PARAMS_CHECK && (base == NULL)) {
+        ucs_error("base memory handle is not provided for memory window");
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    ucs_assertv(!(base->super.flags & UCT_IB_MEM_FLAG_MEM_WINDOW),
+                "memh=%p is already a memory window", base);
```

Comment:
we  should mention this limitation in the API

Reply:
ok